app/javascript/components/main-menu/main-menu.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from 'patternfly-react';
+import TopLevel from './top-level';
+import SecondLevel from './second-level';
+import ThirdLevel from './third-level';
+import { menuProps, RecursiveMenuProps } from './recursive-props';
+
+const getLevelComponent = level => ({
+  0: props => <TopLevel level={level} {...props} />,
+  1: props => <SecondLevel level={level} {...props} />,
+  2: props => <ThirdLevel level={level} {...props} />,
+})[level];
```

Comment:
I think we should add a fallback component if somebody passes level value other than 0 - 3

```jsx
const getLevelComponent = level => ({
  0: props => <TopLevel level={level} {...props} />,
  1: props => <SecondLevel level={level} {...props} />,
  2: props => <ThirdLevel level={level} {...props} />,
})[level] ||  props => <ThirdLevel level={level} {...props} />;
```

This will always return ThirdLevel component if you gave the function any level that it does not cover.